application/app/src/main/java/com/example/buber/Views/Activities/LoginActivity.java

```diff
@@ -32,19 +45,56 @@ protected void onCreate(Bundle savedInstanceState) {
 
         App.getModel().addObserver(this);
 
-        usernameEditText = findViewById(R.id.loginUsernameEditText);
-        passwordEditText = findViewById(R.id.loginPasswordEditText);
-        driverLoginButtonId = R.id.loginDriverButton;
+        editEmail = findViewById(R.id.loginEmailEditText);
+        editPassword = findViewById(R.id.loginPasswordEditText);
+        driverLoginBtnID = R.id.loginDriverButton;
+
+        mFirebaseAuth = FirebaseAuth.getInstance();
+        mAuthStateListener = new FirebaseAuth.AuthStateListener() {
+            @Override
+            public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
+                FirebaseUser mFirebaseUser = mFirebaseAuth.getCurrentUser();
+                if (mFirebaseUser != null) {
+                    Toast.makeText(LoginActivity.this, "You are STILL logged in", Toast.LENGTH_SHORT).show();
+                    // move to the next intent
+                } else {
+                    Toast.makeText(LoginActivity.this, "Please Login", Toast.LENGTH_SHORT).show();
+                }
+            }
+        };
     }
 
```

Comment:
I am not too sure why this is done here. And in your PR u mentioned that u weren't sure about FB user and session user. They're not the same! I was also very confused until Evan told me.

Side note: Is toast a good way of telling a user about an error? or is this just for testing? I ask because on assign 1 the TA said it wasn't.